ast/builtins.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2016 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package ast
+
+// Builtins is the registry of built-in functions supported by
+// OPA. When adding a new built-in function to OPA, update this
+// list.
+var Builtins = [...]*Builtin{
+	Equality,
+}
+
+// BuiltinMap provides a convenient mapping of built-in names to
+// built-in definitions.
+var BuiltinMap map[Var]*Builtin
+
+// Equality represents the "=" operator/function.
+var Equality = &Builtin{
+	Name:         Var("="),
+	NumArgs:      2,
+	RecTargetPos: []int{0, 1},
+}
```

Comment:
My thinking was the built-in function implementations would be specific to topdown. It doesn't make sense to make the AST depend on the topdown package, so I've split the declaration from the implementation. The declaration lives in the  AST package and the implementation lives in the topdown package.